llvm/lib/Transforms/Scalar/DeadStoreElimination.cpp

```diff
@@ -1274,7 +1274,8 @@ struct DSEState {
       if (CB->onlyAccessesInaccessibleMemory())
         return false;
 
-    return isRefSet(BatchAA.getModRefInfo(UseInst, DefLoc));
+    AAQueryInfo TmpAAQI(AA, &EI);
```

Comment:
This shouldn't be necessary. `AA.getModRefInfo(UseInst, DefLoc)` should work.

Reply:
Ah, nice, thanks!

Reply:
Is there a better way to call `getModeRefInfo()` and using `EI` for the query, without having to use the temporary `AAQueryInfo` ?